java/src/main/java/org/hyperledger/fabric/client/Checkpointer.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright 2022 IBM All Rights Reserved.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.hyperledger.fabric.client;
+
+/**
+ * Checkpointer allows update of a checkpoint position after events are successfully processed.
+ */
+public interface Checkpointer extends Checkpoint {
+
+     /**
+      * CheckpointBlock checkpoints block.
+      * @param blockNumber
```

Comment:
Need both method and parameter descriptions.
```suggestion
      * Checkpoint a successfully processed block.
      * <p>Note that the block number is an unsigned 64-bit integer, with the sign bit used to hold the top bit of
      * the number.</p>
      * @param blockNumber a ledger block number.
```